0011-R-multiple_reminders_actors_impl.md

```diff
@@ -0,0 +1,149 @@
+
+	SetHaltActorFns(haltFn HaltActorFn, haltAllFn HaltAllActorsFn)
+	SetOnAPILevelUpdate(fn func(apiLevel uint32))
+
+	// PlacementHealthy returns true if the placement service is healthy.
+	PlacementHealthy() bool
+	// StatusMessage returns a custom status message.
+	StatusMessage() string
+}
+
+```
+
+```go
+type RemindersProvider interface {
+	io.Closer
+
+	Init(ctx context.Context) error
+	GetReminder(ctx context.Context, req *GetReminderRequest) (*Reminder, error)
+	CreateReminder(ctx context.Context, req *Reminder) error
+	DeleteReminder(ctx context.Context, req DeleteReminderRequest) error
+	DrainRebalancedReminders(actorType string, actorID string)
+	OnPlacementTablesUpdated(ctx context.Context)
+
+	SetResiliencyProvider(resiliency resiliency.Provider)
+	SetExecuteReminderFn(fn ExecuteReminderFn)
+	SetStateStoreProviderFn(fn StateStoreProviderFn)
+	SetLookupActorFn(fn LookupActorFn)
+}
+
+```
+Idea behind this is that interfaces defined in pkg/actors/internal will be used to form a Building Block / Client which will implement all these methods defined in these interfaces.
+
+Whereas, methods defined for Services in protos will be implemented by respective new/existing Control Plane Service(s).
+
+And, the methods from these clients will call methods of the servers, wherever required.
+
+![actors_reminders](./resources/0011-R-multiple_reminders_actors_impl/Actors_Reminders_Interface.png)
+
+Depending on the implementation, it can be decided that what all RPCs need to be defined for a Service. Example, for a Placement Service which just creates and pushes Actor Types v/s Dapr Hosts table information to All sidecars, only table related ReportDaprStatusneeds to be as rpc.
+
+Whereas, for a Service, where Actor information is Pulled by sidecars from Actors Service needs some methods like LookupActor or ReportActorDeactivation, which thus form a part of RPCs.
+
+For some minor changes left in interfaces,  has been raised (https://github.com/dapr/dapr/pull/7281).
+
+## How to build and deploy Actors/Scheduler/Placement Service?
+
+There are two sub-systems that we are dealing with here: Actors and Reminders. And, as we know, going forward, there can be multiple variants of Services providing either of these sub-systems or both.
+
+There will be multiple possibilities, like:
+
+1. Placement Service used for Actors and Distributed Scheduler used for Reminders.
+
+2. ABC Service used for both Actors and Reminders.
+
+3. XYZ used for Actors and Distributed Scheduler used for Reminders.
+
+So, how to decide what all to deploy in what case for user?
+
+### Steps:
+
+1. Dapr Configuration changes to depict what services to load for Control Plane and what not.
+
+2. Component loading capability, specifically by operator, for Control Plane. This is required, in case CP requires a component to be loaded.
+
+#### Step 1: Configuration changes:
+
+A) Changes in charts/dapr/charts/dapr_config/templates/dapr_default_config.yaml
```

Comment:
I don't think a vendor's implementation should impact the OSS Chart. It is the vendor's responsibility to provide the Chart for their own services. Otherwise, each vendor will need to PR against the OSS for any specific of their custom implementation.

Reply:
Yeah, it is not affected. I will remove it from list of updates. https://github.com/dapr/dapr/pull/7318 doesn't add any such change.